src/gen9_hevc_enc_kernels.h

```diff
@@ -0,0 +1,2318 @@
+} gen9_hevc_enc_kernels_header;
+
+typedef struct _gen9_hevc_enc_kernels_header_bxt {
+    int kernel_count;
+    gen9_hevc_enc_kernel_header HEVC_ENC_I_2xDownSampling_Kernel;
+    gen9_hevc_enc_kernel_header HEVC_ENC_I_32x32_PU_ModeDecision_Kernel;
+    gen9_hevc_enc_kernel_header HEVC_ENC_I_16x16_PU_SADComputation_Kernel;
+    gen9_hevc_enc_kernel_header HEVC_ENC_I_16x16_PU_ModeDecision_Kernel;
+    gen9_hevc_enc_kernel_header HEVC_ENC_I_8x8_PU_Kernel;
+    gen9_hevc_enc_kernel_header HEVC_ENC_I_8x8_PU_FMode_Kernel;
+    gen9_hevc_enc_kernel_header HEVC_ENC_PB_32x32_PU_IntraCheck;
+    gen9_hevc_enc_kernel_header HEVC_ENC_PB_MB;
+    gen9_hevc_enc_kernel_header HEVC_ENC_I_DS4HME;
+    gen9_hevc_enc_kernel_header HEVC_ENC_P_HME;
+    gen9_hevc_enc_kernel_header HEVC_ENC_B_HME;
+    gen9_hevc_enc_kernel_header HEVC_ENC_I_COARSE;
+    gen9_hevc_enc_kernel_header HEVC_ENC_BRC_Init;
+    gen9_hevc_enc_kernel_header HEVC_ENC_BRC_Reset;
+    gen9_hevc_enc_kernel_header HEVC_ENC_BRC_Update;
+    gen9_hevc_enc_kernel_header HEVC_ENC_BRC_LCU_Update;
+    gen9_hevc_enc_kernel_header HEVC_ENC_PB_Pak;
+    gen9_hevc_enc_kernel_header HEVC_ENC_PB_Widi;
+    gen9_hevc_enc_kernel_header HEVC_ENC_BRC_Blockcopy;
+    gen9_hevc_enc_kernel_header HEVC_ENC_DS_Combined;
+    gen9_hevc_enc_kernel_header HEVC_ENC_P_MB;
+    gen9_hevc_enc_kernel_header HEVC_ENC_P_Widi;
+} gen9_hevc_enc_kernels_header_bxt;
+
+typedef enum _GEN9_HEVC_ENC_MBENC_KRNIDX
+{
+    GEN9_HEVC_ENC_MBENC_2xSCALING = 0,
+    GEN9_HEVC_ENC_MBENC_32x32MD,
+    GEN9_HEVC_ENC_MBENC_16x16SAD,
+    GEN9_HEVC_ENC_MBENC_16x16MD,
+    GEN9_HEVC_ENC_MBENC_8x8PU,
+    GEN9_HEVC_ENC_MBENC_8x8FMODE,
+    GEN9_HEVC_ENC_MBENC_32x32INTRACHECK,
+    GEN9_HEVC_ENC_MBENC_BENC,
+    GEN9_HEVC_ENC_MBENC_BPAK,
+    GEN9_HEVC_ENC_MBENC_WIDI,
+    GEN9_HEVC_ENC_MBENC_NUM,
+    GEN9_HEVC_ENC_MBENC_DS_COMBINED = GEN9_HEVC_ENC_MBENC_NUM,
+    GEN95_HEVC_ENC_MBENC_NUM_KBL,
+    GEN9_HEVC_MBENC_PENC = GEN95_HEVC_ENC_MBENC_NUM_KBL,
+    GEN9_HEVC_MBENC_P_WIDI,
+    GEN9_HEVC_MBENC_NUM_BXT,
+    GEN8_HEVC_ENC_MBENC_TOTAL_NUM = GEN9_HEVC_MBENC_NUM_BXT
+} GEN9_HEVC_ENC_MBENC_KRNIDX;
+
+typedef enum _GEN9_HEVC_ENC_BRC_KRNIDX
+{
+    GEN9_HEVC_ENC_BRC_COARSE_INTRA = 0,
+    GEN9_HEVC_ENC_BRC_INIT,
+    GEN9_HEVC_ENC_BRC_RESET,
+    GEN9_HEVC_ENC_BRC_FRAME_UPDATE,
+    GEN9_HEVC_ENC_BRC_LCU_UPDATE,
+    GEN9_HEVC_ENC_BRC_NUM
+} GEN9_HEVC_ENC_BRC_KRNIDX;
+
+typedef enum _GEN9_ENC_ENC_OPERATION
+{
+    GEN9_ENC_SCALING4X = 0,
+    GEN9_ENC_SCALING2X,
+    GEN9_ENC_ME,
+    GEN9_ENC_BRC,
+    GEN9_ENC_MBENC,
+    GEN9_ENC_MBENC_WIDI,
+    GEN9_ENC_RESETVLINESTRIDE,
+    GEN9_ENC_MC,
+    GEN9_ENC_MBPAK,
+    GEN9_ENC_DEBLOCK,
+    GEN9_ENC_PREPROC,
+    GEN9_ENC_VDENC_ME,
+    GEN9_ENC_WP,
+    GEN9_ENC_SFD,
+    GEN9_ENC_MBENC_I_LUMA,
+    GEN9_ENC_MPU,
+    GEN9_ENC_TPU,
+    GEN9_ENC_SCALING_CONVERSION,
+    GEN9_ENC_DYS
+} GEN9_ENC_OPERATION;
+
+#define GEN9_HEVC_ENC_KERNEL_SIZE 147456
```

Comment:
the  structure definition should be defined in one dedicated file while the kernel binary is defined in another file.